pages/[host]/[collection]/[tokenManagerString].tsx

```diff
@@ -23,15 +32,64 @@ export async function getServerSideProps(context: any) {
     commitment: 'recent',
   })
 
-  const tokenManagerId = tryPublicKey(tokenManagerString)
-  if (!tokenManagerId) {
+  let mintId
+  let isClaimed
+  if (mintIdString) {
+    mintId = tryPublicKey(mintIdString)
+    isClaimed = false
+  } else {
+    const tokenManagerId = tryPublicKey(tokenManagerString)
+    if (!tokenManagerId) {
+      return {}
+    }
+    const tokenManagerData = await tokenManager.accounts.getTokenManager(
+      connection,
+      tokenManagerId
+    )
+    mintId = tokenManagerData.parsed.mint
+    isClaimed = tokenManagerData.parsed.state === TokenManagerState.Claimed
+  }
+
+  if (!mintId) {
     return {}
   }
-  const tokenData = await getTokenData(connection, tokenManagerString)
+  const [metaplexId] = await metaplex.MetadataProgram.findMetadataAccount(
+    mintId
+  )
+  const metaplexDataRaw = await metaplex.Metadata.load(
+    connection,
+    metaplexId
+  ).catch((e) => {
+    console.log('Failed to get metaplex data', e)
+    return null
+  })
+  const metaplexData = metaplexDataRaw
+    ? {
+        pubkey: metaplexDataRaw?.pubkey,
+        parsed: metaplexDataRaw?.data,
+      }
+    : undefined
+
+  let metadata: AccountData<any> | null = null
+  if (metaplexData) {
+    try {
+      const json = await fetch(metaplexData.parsed.data.uri).then((r) =>
+        r.json()
+      )
+      metadata = { pubkey: metaplexData.pubkey, parsed: json }
+    } catch (e) {
+      console.log('Failed to get metadata data', e)
+    }
+  }
```

Comment:
this works though i wouldve just done like imageUrl = json.image instead of like putting it into this AccountData type for no reason haha